src/main/java/ee/carlrobert/codegpt/settings/GeneralSettings.java

```diff
@@ -41,8 +41,18 @@ public static GeneralSettings getInstance() {
   public void sync(Conversation conversation) {
     var clientCode = conversation.getClientCode();
     if ("chat.completion".equals(clientCode)) {
-      state.setSelectedService(ServiceType.OPENAI);
-      OpenAISettings.getCurrentState().setModel(conversation.getModel());
+      switch (state.getSelectedService()) {
```

Comment:
You are entirely correct, I am not sure how I managed to reproduce this locally. I am guessing that something interfered with persisting my settings, and thus I thought the service was switched back. I am closing this PR since these code changes are wrong.

Reply:
The bug is still valid tho. Try this test-case:
```kt
fun testCustomOpenAISettingsSync() {
    val settings = service<GeneralSettings>().apply {
        state.selectedService = ServiceType.OPENAI
    }

    settings.sync(Conversation().apply { clientCode = "custom.openai.chat.completion" })

    assertThat(settings.state.selectedService).isEqualTo(ServiceType.CUSTOM_OPENAI)
}
```

Reply:
Yes, I submitted a separate PR #472 to fix that.